bin/dotd_menu

```diff
@@ -0,0 +1,349 @@
+
+@logger.formatter = proc do |_serverity, time, _progname, msg|
+  "[#{time.strftime('%I:%M %p')}]: #{msg}\n"
+end
+
+# Checks that CDO.github_access_token and CDO.slack_token are defined. If not,
+# exits the program.
+def check_for_cdo_keys
+  return if CDO.github_access_token && CDO.slack_token && CDO.honeybadger_api_token
+
+  puts <<-EOS.unindent
+
+    This script requires CDO.github_access_token, CDO.slack_token, and CDO.honeybadger_api_token.
+
+    Create your API tokens from these pages:
+
+      https://github.com/settings/tokens ('public_repo' permission)
+      https://api.slack.com/custom-integrations/legacy-tokens
+      https://app.honeybadger.io/users/edit#authentication
+
+    Please add them to your locals.yml and rerun the script.
+
+  EOS
+  exit
+end
+
+# Prints the script intro to the console.
+def puts_script_intro
+  # TODO(asher): Update this appropriately.
+
+  puts <<-EOS.unindent
+
+    #{bold 'Welcome to Robo-Dev-of-the-Day!'}
+
+    #{bold 'Documentation'}
+
+      Dev-of-the-Day Checklist
+      #{dim 'http://wiki.code.org/display/PROD/Dev+of+the+Day+Checklist'}
+
+      Known Issues
+      #{dim 'https://docs.google.com/document/d/1zaHwuosda-6YYCkNdTFG7gw_iMHTm_BYCNlf6UqEJLg/edit#'}
+
+  EOS
+end
+
+def mark_dtp_finished
+  DevelopersTopic.set_dtp 'yes'
+  InfraProductionTopic.set_dtp_commit GitHub.sha('production')
+end
+
+def mark_dtt_green
+  DevelopersTopic.set_dtt 'yes'
+  InfraTestTopic.set_green_commit GitHub.sha('test')
+end
+
+def do_dtp
+  test_green_commit = InfraTestTopic.green_commit
+  DevelopersTopic.set_dtp "in progress (@#{@dotd_name})"
+
+  # As GitHub's API (all of ours, Octokit's and github.com's) does not allow creating a pull
+  # request from a commit hash (only a branch name), we create and push a branch.
+  branch_name = "dtp_candidate_#{test_green_commit}"
+  begin
+    create_branch_from_commit branch_name, test_green_commit
+    production_pr_number = GitHub.create_pull_request(
+      base: 'production',
+      head: branch_name,
+      title: "DTP (Test > Production: #{test_green_commit})"
+    )
+    print_database_changes production_pr_number
+    dtp_auto_merged = should_i "merge DTP PR#{production_pr_number}" do
+      GitHub.merge_pull_request(
+        production_pr_number,
+        "DTP (Test > Production: #{test_green_commit})"
+      )
+    end
+    GitHub.open_url GitHub.url(production_pr_number) unless dtp_auto_merged
+    delete_branch branch_name
+  end
+end
+
+# Repeatedly prints a menu of DOTD tasks, accepts a choice from the user, and executes the choice.
+def dotd_menu
+  begin
+    loop do
+      puts <<-EOS.unindent
+
+        #{bold 'DOTD Menu Options'}
+
+          Honeybadger:
+
+            #{bold 'H'}: Retrieve Honeybadger issues.
+
+          DTT:
+
+            #{bold 'G'}: Mark the most recent DTT green.
+
+          DTP:
+
+            #{bold 'P'}: DTP the most recent green DTT.
+            #{bold 'F'}: Mark the most recent DTP finished.
```

Comment:
:+1: Sounds good.  I guess I meant more generally, I'd eventually like this to be more context-sensitive regarding suggested/available actions.  But maybe that work can wait until we've got a proper bot set up. :grin: